external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala

```diff
@@ -56,7 +56,7 @@ private[avro] class AvroFileFormat extends FileFormat with DataSourceRegister {
       spark: SparkSession,
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
-    val conf = spark.sparkContext.hadoopConfiguration
+    val conf = spark.sessionState.newHadoopConf()
```

Comment:
A good catch!

Reply:
can we add a linter rule for this?

Reply:
Sure, will do it.
Can I make it a simple one, check the appearance  of 
`= spark.sparkContext.hadoopConfiguration` ?
Otherwise there are too many usage for the `sparkContext.hadoopConfiguration`